core/service/izanami.cpp

```diff
@@ -22,160 +22,216 @@ limitations under the License.
 #include "executor.hpp"
 #include <infra/protobuf/api.pb.h>
 #include <consensus/connection/connection.hpp>
+#include <service/peer_service.hpp>
 #include <infra/config/peer_service_with_json.hpp>
 #include <infra/config/iroha_config_with_json.hpp>
 #include <crypto/hash.hpp>
 #include <repository/transaction_repository.hpp>
 
+namespace peer {
+    namespace izanami {
+        using Api::TransactionResponse;
 
-namespace izanami {
-    using Api::TransactionResponse;
 
-    InitializeEvent::InitializeEvent() {
-        now_progress = 0;
-    }
+        void setAwkTimer(int const sleepMillisecs, const std::function<void(void)> &action) {
+            std::thread([action, sleepMillisecs]() {
+                std::this_thread::sleep_for(std::chrono::milliseconds(sleepMillisecs));
+                action();
+            }).join();
```

Comment:
We refactor `utils/`, genreral function will migrate to there.
Would you wait ?